casper/src/main/scala/coop/rchain/casper/util/comm/CommUtil.scala

```diff
@@ -77,12 +88,15 @@ object CommUtil {
             local    <- TransportLayer[F].local
             block    = dag.blockLookup.get(r.hash).map(_.toByteString)
             maybeMsg = block.map(serializedMessage => packet(local, serializedMessage))
-            send     <- maybeMsg.traverse(msg => TransportLayer[F].send(peer, msg))
+            send     <- maybeMsg.traverse(msg => TransportLayer[F].roundTrip(peer, msg, defaultTimeout))
```

Comment:
This seems like a strange implementation to me. The communication originates with the other node sending the request, so I would expect the `roundTrip` to be called there and the response to be given directly here, as opposed to starting another `roundTrip`. But maybe it makes sense because then both sides get to know their messages were received?

Reply:
@birchmd: Do you want to keep the *send and forget* semantic here?
@rabbitonweb ?

Reply:
The logic is as follows. Suppose we have two nodes A and B; A sends a BlockRequest message to B; B recieves the message and checks if it has the block, if so then sends the block to A. Now, what this should look like in terms of communications functions I'm not sure. I could see this as a single `roundTrip` initiated by A, where the block from B comes as a response. Or, as it is right now, with two `roundTrips` where two additional messages are sent, which are effectively just acknowledgements.

I suppose having the extra two acknowledgements isn't a problem really, so maybe it's fine as is.

Reply:
Let's keep the message passing style with acknowledgements.

Reply:
@sebtomba
> I suppose having the extra two acknowledgements isn't a problem really, so maybe it's fine as is.

in the scenario @birchmd  described, what happens when block A don't send the `ack` in time, does B initiates another `roundTrip` ?